src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -3068,6 +3068,150 @@ public override MemberInfo[] GetMember(string name, MemberTypes type, BindingFla
 
             return compressMembers;
         }
+
+        public override MemberInfo GetMemberWithSameMetadataDefinitionAs(MemberInfo member)
+        {
+            if (member is null) throw new ArgumentNullException(nameof(member));
+
+            MemberInfo? result = member.MemberType switch
+            {
+                MemberTypes.Method => GetMethodWithSameMetadataDefinitionAs(this, member),
+                MemberTypes.Constructor => GetConstructorWithSameMetadataDefinitionAs(this, member),
+                MemberTypes.Property => GetPropertyWithSameMetadataDefinitionAs(this, member),
+                MemberTypes.Field => GetFieldWithSameMetadataDefinitionAs(this, member),
+                MemberTypes.Event => GetEventWithSameMetadataDefinitionAs(this, member),
+                MemberTypes.NestedType => GetNestedTypeWithSameMetadataDefinitionAs(this, member),
+                _ => null
+            };
+
+            return result ?? throw CreateGetMemberWithSameMetadataDefinitionAsNotFoundException(member);
+        }
+
+        private static MemberInfo? GetMethodWithSameMetadataDefinitionAs(RuntimeType? runtimeType, MemberInfo method)
+        {
+            while (runtimeType != null)
```

Comment:
Can this loop that walks the inheritance hiearchy be in GetMemberWithSameMetadataDefinitionAs to reduce code duplication?

Reply:
Fixed.